### PR TITLE
change emission mechanism

### PR DESCRIPTION
**Pull request by agiscientist**
1⃣️ Introduce burned mechanism.

```bash
2025-03-03 12:32:53.845 |       INFO       | bittensor:evaluator.py:354 | Text embedding quality loss: 0.7242655158042908
2025-03-03 12:32:55.449 |       INFO       | bittensor:evaluator.py:354 | Text embedding quality loss: 0.726318359375
2025-03-03 12:32:55.461 |       INFO       | bittensor:validator.py:456 | Allocated 90.0% incentive to burner miner UID 165
2025-03-03 12:32:55.479 |       INFO       | bittensor:validator.py:457 | All rewards: tensor([0.1000, 0.9000])
2025-03-03 12:32:55.479 |      DEBUG       | bittensor:validator.py:316 | Scattered rewards: tensor([0.1000, 0.9000])
2025-03-03 12:32:55.488 |      DEBUG       | bittensor:validator.py:324 | Updated moving avg scores: tensor([0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 7.9028e-02, 2.6016e-08,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00,
        0.0000e+00, 0.0000e+00, 0.0000e+00, 7.1125e-01])
```

2⃣️ It seems that Hugging Face now requires login to access FineWeb/Wikipedia. If you encounter issues loading the data, please ensure you have the necessary authentication. You can find your Hugging Face token at: https://huggingface.co/settings/tokens.

```bash
huggingface-cli login
```

### CODE REVIEW

File: neurons/validator.py [hunk 1]
```diff
@@ -308,19 +323,38 @@ async def forward(self):
...
+                # Calculate total reward
+                total_reward = rewards.sum()
+                
+                # Create a new rewards tensor that includes the burned miner
+                all_uids = torch.cat([miner_uids, torch.tensor([burned_miner_uid])])
+                all_rewards = torch.zeros(len(all_uids))
+                
+                # Allocate 10% of rewards proportionally to the randomly selected miners
+                if total_reward > 0:
+                    all_rewards[:len(miner_uids)] = rewards * 0.1
```
Comment (webroot-kaito) [inline neurons/validator.py:343]: Let's set the reward to be a parameter that we pull from Gitbub (also the wallet address) so that the update can be instant, similar to how dataset rotation is implemented. 

https://github.com/OpenKaito/openkaito/blob/main/datasets_config.json

**Review by webroot-kaito — COMMENTED** [review]
Can we add a test plan with screen shot showing that our two miners receive the expected emission split from the testnet?

File: neurons/validator.py (hunk 1 above)
Comment (agiscientist) [reply neurons/validator.py]: solved

File: neurons/validator.py [hunk 2]
```diff
@@ -379,13 +403,50 @@ async def forward(self):
                 rewards = torch.zeros(len(miner_uids))
 
             raw_scores = rewards.clone().detach()
-
-            # relative scores in a batch
             rewards = rewards / (rewards.max() + 1e-5)
 
-            bt.logging.info(f"Scored responses: {rewards} for {miner_uids}")
+            if burned_miner_uid is not None:
```
Comment (webroot-kaito) [inline neurons/validator.py:408]: nit: let's call it "burner_miner" instead of "burned_miner"

File: neurons/validator.py [hunk 3]
```diff
@@ -379,13 +403,50 @@ async def forward(self):
...
+                total_reward = rewards.sum()
+                
+                # Check if burned miner is in the selected miners
+                burned_in_selection = burned_miner_uid in miner_uids
+                
+                if burned_in_selection:
+                    bt.logging.info(f"Burned miner UID {burned_miner_uid} is in the randomly selected miners.")
+                    # Get the index of burned_miner_uid in miner_uids
+                    burned_idx_in_random = miner_uids.tolist().index(burned_miner_uid)
+                else:
```
Comment (yang-han) [inline neurons/validator.py:418]: actually we may enforce the burner miner to be selected in the miner selection stage (e.g., always select (k-1) miners from all miners except the burner, then add the burner.

Comment (agiscientist) [reply neurons/validator.py]: previous is right, but to keep consistent, I follow the discussed setting now

**Review by webroot-kaito — APPROVED** [review]
lgtm!
